packages/loot-core/src/server/accounts/sync.ts

```diff
@@ -250,54 +250,12 @@ async function normalizeBankSyncTransactions(transactions, acctId) {
       throw new Error('`date` is required when adding a transaction');
     }
 
-    let payee_name;
-    // When the amount is equal to 0, we need to determine
-    // if this is a "Credited" or "Debited" transaction. This means
-    // that it matters whether the amount is a positive or negative zero.
-    if (trans.amount > 0 || Object.is(Number(trans.amount), 0)) {
-      const nameParts = [];
-      const name =
-        trans.debtorName ||
-        trans.remittanceInformationUnstructured ||
-        (trans.remittanceInformationUnstructuredArray || []).join(', ') ||
-        trans.additionalInformation;
-
-      if (name) {
-        nameParts.push(title(name));
-      }
-      if (trans.debtorAccount && trans.debtorAccount.iban) {
-        nameParts.push(
-          '(' +
-            trans.debtorAccount.iban.slice(0, 4) +
-            ' XXX ' +
-            trans.debtorAccount.iban.slice(-4) +
-            ')',
-        );
-      }
-      payee_name = nameParts.join(' ');
-    } else {
-      const nameParts = [];
-      const name =
-        trans.creditorName ||
-        trans.remittanceInformationUnstructured ||
-        (trans.remittanceInformationUnstructuredArray || []).join(', ') ||
-        trans.additionalInformation;
-
-      if (name) {
-        nameParts.push(title(name));
-      }
-      if (trans.creditorAccount && trans.creditorAccount.iban) {
-        nameParts.push(
-          '(' +
-            trans.creditorAccount.iban.slice(0, 4) +
-            ' XXX ' +
-            trans.creditorAccount.iban.slice(-4) +
-            ')',
-        );
-      }
-      payee_name = nameParts.join(' ');
+    if (trans.payeeName == null) {
+      throw new Error('`payeeName` is required when adding a transaction');
     }
```

Comment:
This will break SimpleFIN support as it does not provide this field.

Reply:
<img width="729" alt="image" src="https://github.com/actualbudget/actual/assets/1115390/36386eca-aa22-4119-b2a0-0d7aa127e86e">

Reply:
Note that SimpleFIN seems to provide `payee` as a field.  I am not familiar with the GoCardless side, but could it just be renamed to `payee` instead of `payeeName`?

Reply:
Maybe the issue is that the SimpleFIN server code needs to call the moved function and generate a `payeeName`?

Reply:
I'm not sure if we'd want to overwrite an field already provided by the APIs, so I added `payeeName` to the simplefin `/transactions` endpoint in the actual-server complimentary PR. Could you try running this PR with the server PR (actualbudget/actual-server#353) to confirm if this issue persists?

https://github.com/actualbudget/actual-server/blob/ffdc9684201d3d342dcabf75f23ad8532426b81b/src/app-simplefin/app-simplefin.js#L112

Happy to switch approaches if needed?

Reply:
I think you are on the right track.  Shouldn't that new `payeeName` also call the function that was moved to the server side?

Reply:
So something like `newTrans.payeeName = formatPayeeName(trans);` instead.

Reply:
I am not entirely sure the SimpleFIN transactions will actually change the name via that function, but at least the same code path will run as it does currently.

Reply:
Yes, definitely - good spot. Will get the util moved down a directory and call from both.

Reply:
That might also imply the function should live in another folder, instead of under the GoCardless code, since it is shared across two different bank syncs.

Reply:
Should be fixed in `actual-server` now, util moved up a directory and run on the simplefin transacion